crates/block-producer/src/psc.rs

```diff
@@ -736,6 +766,73 @@ async fn median_gte(rpc_client: &RPCClient, timestamp_millis: u64) -> Result<()>
     Ok(())
 }
 
+/// Submit next pending l1 upgrade
+async fn submit_pending_l1_upgrade(ctx: &PSCContext) -> Result<()> {
+    let snap = ctx.store.get_snapshot();
+    let block_number = snap
+        .get_last_confirmed_block_number_hash()
```

Comment:
`get_last_submitted_block_number_hash()`?

Target `l1_upgrade.height` may be submitted.

Reply:
This function is called just before the `submit_next_block`, so `height + 1` block hasn't be submitted yet. https://github.com/godwokenrises/godwoken/pull/971/files#diff-b91893c4ad9a6897506ded8d9e15175017791e9aa4f256b093d0e1402da23bc0R261

Reply:
Suppose we want to upgrade at height 100 (`l1_upgrade.height = 100`). 

The code only works as expected when 99 is confirmed but 100 is not submitted. Since we submit 3 blocks in chained txs by default. 100 aka `height + 1` may be submitted.

Reply:
Ok, we should use `get_last_submitted_block_number_hash` here